lib/mongoose-core.module.ts

```diff
@@ -59,7 +65,13 @@ export class MongooseCoreModule implements OnApplicationShutdown {
               await mongoose.createConnection(uri, mongooseOptions).asPromise(),
               mongooseConnectionName,
             ),
-          ).pipe(handleRetry(retryAttempts, retryDelay)),
+          ).pipe(
+            handleRetry(retryAttempts, retryDelay),
+            catchError((error) => {
+              mongooseConnectionError(error);
+              throw error;
```

Comment:
```suggestion
              throw mongooseConnectionError(error);
```
what about this approach instead (this in combination with https://github.com/nestjs/mongoose/pull/1550/files#r1003070783)?

Reply:
That's a very nice suggestion